examples/30_extended/custom_flow_tutorial.py

```diff
@@ -0,0 +1,202 @@
+subflow = dict(components=OrderedDict(automl_tool=autosklearn_flow),)
+
+####################################################################################################
+# With all parameters of the flow defined, we can now initialize the OpenMLFlow and publish.
+# Because we provided all the details already, we do not need to provide a `model` to the flow.
+#
+# In our case, we don't even have a model. It is possible to have a model but still require
+# to follow these steps when the model (python object) does not have an extensions from which
+# to automatically extract the hyperparameters.
+# So whether you have a model with no extension or no model at all, explicitly set
+# the model of the flow to `None`.
+
+autosklearn_amlb_flow = openml.flows.OpenMLFlow(
+    **general, **flow_hyperparameters, **subflow, model=None,
+)
+autosklearn_amlb_flow.publish()
+print(f"autosklearn flow created: {autosklearn_amlb_flow.flow_id}")
+
+####################################################################################################
+# 2. Using the flow
+# ====================
+# This Section will show how to upload run data for your custom flow.
+# Take care to change the values of parameters as well as the task id,
+# to reflect the actual run.
+# Task and parameter values in the example are fictional.
+
+flow_id = autosklearn_amlb_flow.flow_id
+
+parameters = [
+    OrderedDict([("oml:name", "cores"), ("oml:value", 4), ("oml:component", flow_id)]),
+    OrderedDict([("oml:name", "memory"), ("oml:value", 16), ("oml:component", flow_id)]),
+    OrderedDict([("oml:name", "time"), ("oml:value", 120), ("oml:component", flow_id)]),
+]
+
+task_id = 1408  # Iris Task
+task = openml.tasks.get_task(task_id)
+dataset_id = task.get_dataset().dataset_id
+
+
+####################################################################################################
+# The last bit of information for the run we need are the predicted values.
+# The exact format of the predictions will depend on the task.
+#
+# The predictions should always be a list of lists, each list should contain:
+# - the repeat number: for repeated evaluation strategies. (e.g. repeated cross-validation)
+# - the fold number: for cross-validation. (what should this be for holdout?)
+# - 0: this field is for backward compatibility.
+# - index: the row (of the original dataset) for which the prediction was made.
+# - p_1, ..., p_c: for each class the predicted probability of the sample
+#   belonging to that class. (no elements for regression tasks)
+#   Make sure the order of these elements follows the order of `task.class_labels`.
+# - the predicted class/value for the sample
+# - the true class/value for the sample
+#
+# When using openml-python extensions (such as through `run_model_on_task`),
+# all of this formatting is automatic.
+# Unfortunately we can not automate this procedure for custom flows,
+# which means a little additional effort is required.
```

Comment:
For me personally, the minor gap in understanding that exists is the behaviour of functions available in the OpenML API. Given that the example so far creates a *custom* flow, pulls a task and a dataset and then jumps to how the predictions should be formatted for a legitimate OpenML run format, leaves me wondering whether I can use any OpenML function to reach till the prediction formatting stage. 
As it appears, one would have to pull the dataset using *task* API such as `get_X_and_y`, and explicitly call fit(), predict() or the equivalent, and then format the output predictions as you mention later in the example.

If I have understood it correctly, then I think that having at least comments talking briefly about the next imminent steps to go from flow, task, data to predictions might be helpful. Though on second thought, what I said feels like quite the obvious next steps when dealing with such a flow, maybe being explicit might help the new OpenML users. 

To summarise, the fact that `run_model_on_task` may not work on a custom flow without a registered extension could be more explicit, if that is indeed what is supposed to happen.

Reply:
> As it appears, one would have to pull the dataset using task API such as get_X_and_y, and explicitly call fit(), predict() or the equivalent, and then format the output predictions as you mention later in the example.

Correct, as custom flows are used when OpenML can't interface directly with the machine learning flow.

> If I have understood it correctly, then I think that having at least comments talking briefly about the next imminent steps to go from flow, task, data to predictions might be helpful.

I am not sure what you are missing here, all of this is documented step by step in this tutorial? Exactly how to feed data to the model and produce predictions is dependent on the framework you work with. In our case we just generate predictions instead, but in general we don't know the interface which is why it's not available as an extension in the first place.

> To summarise, the fact that run_model_on_task may not work on a custom flow without a registered extension could be more explicit, if that is indeed what is supposed to happen.

In which way do you feel the opening paragraphs do not address this? (I did just notice I forgot to put in the reference link, so perhaps that's why)

Reply:
yes, the link makes it clear :)